### ISSUE TITLE
sf project retrieve start should not require source tracking to retrieve a package

### ISSUE THREAD

**Issue by FabienTaillon** [forcedotcom/cli#2091]
### Summary
I've been using `sfdx force:source:retrieve -n mypackage` to retrieve a package content from a sandbox, but saw the following warning: ` We plan to deprecate this command in the future. Try using the "project retrieve start" command instead`.

So I tried to move to `sf project retrieve start -n mypackage` but then got the following : `This command can only be used on orgs that have source tracking enabled, such as sandboxes and scratch orgs`. As I just want to retrieve a package content, I don't think source tracking should be required.

### Steps To Reproduce:

**Repository to reproduce:** [dreamhouse-lwc](https://github.com/dreamhouseapp/dreamhouse-lwc)

1.  Go to Setup/Package Manager
2.  Create a Package, for instance named `mypackage`
3. Add whatever metadata you want, like the Dreamhouse custom app
4. From the CLI, try to retrieve the package content using `sf project retrieve start -n mypackage`

### Expected result

The package content is retrieved.

### Actual result

The commands fails with the following error: `This command can only be used on orgs that have source tracking enabled, such as sandboxes and scratch orgs`.

### System Information

- Which shell/terminal are you using? (e.g. bash, zsh, powershell 5, powershell 7, cmd.exe, etc.)
**zsh**

- If you are using `sfdx`
	- Run `sfdx version --verbose --json`
- If you are using `sf`
	- Run `sf version --verbose --json`
- Paste the output here
```
{
  "cliVersion": "@salesforce/cli/1.73.0",
  "architecture": "darwin-arm64",
  "nodeVersion": "node-v18.15.0",
  "osVersion": "Darwin 22.4.0",
  "shell": "zsh",
  "rootPath": "/Users/fabientaillon/.local/share/sfdx/client/7.197.8-077a68b/sf",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.1.8 (core)",
    "@oclif/plugin-commands 2.2.13 (core)",
    "@oclif/plugin-help 5.2.9 (core)",
    "@oclif/plugin-not-found 2.3.23 (core)",
    "@oclif/plugin-plugins 2.4.4 (core)",
    "@oclif/plugin-search 0.0.15 (core)",
    "@oclif/plugin-update 3.1.9 (core)",
    "@oclif/plugin-version 1.3.2 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.33 (core)",
    "@oclif/plugin-which 2.2.18 (core)",
    "@salesforce/cli 1.73.0 (core)",
    "apex 2.2.9 (core)",
    "auth 2.7.11 (core)",
    "data 2.3.9 (core)",
    "deploy-retrieve 1.8.8 (core)",
    "dev 0.7.0 (user)",
    "info 2.6.4 (core)",
    "limits 2.3.12 (core)",
    "login 1.2.5 (core)",
    "org 2.6.5 (core)",
    "packaging 1.15.2 (user)",
    "schema 2.3.6 (core)",
    "settings 1.4.3 (core)",
    "sobject 0.1.14 (core)",
    "source 2.9.1 (core)",
    "telemetry 2.1.3 (core)",
    "templates 55.4.5 (core)",
    "trust 2.4.6 (core)",
    "user 2.3.7 (core)",
    "my-test-plugin 0.0.1 (link) /Users/fabientaillon/Documents/tests/my-test-plugin",
    "texei-sfdx-plugin 2.0.0 (link) /Users/fabientaillon/Documents/code/texei-sfdx-plugin"
  ]
}
```

**Comment by shetzel** [comment]
Reproduced with provided steps using sf v1.76.1.  We'll get it fixed asap.

**Comment by mshanemc** [comment]
done, shipped this week (sfdx and sf)

**Comment by ghost** [comment]
@mshanemc I've been playing around in the V2 beta of the SF CLI, and noticed the same problem happen. I found this issue 2 weeks ago so didn't make a new issue as this should resolve it as well. Trying the same thing now still in v2 seems to continue to have a similar problem. Do you know if this fix has been applied to V2 as well? If so, then I'll dig a little more into it.

Edit:
I have downgraded to the below information and still receive similar problems. If you like I will make a new issue for this as I shouldn't be bothering you on a closed issue :)
```JSON
{
  "cliVersion": "@salesforce/cli/1.79.5",
  "architecture": "darwin-x64",
  "nodeVersion": "node-v18.15.0",
  "osVersion": "Darwin 22.4.0",
  "shell": "zsh",
  "rootPath": "/usr/local/lib/node_modules/@salesforce/cli",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.2.0 (core)",
    "@oclif/plugin-commands 2.2.15 (core)",
    "@oclif/plugin-help 5.2.9 (core)",
    "@oclif/plugin-not-found 2.3.23 (core)",
    "@oclif/plugin-plugins 3.0.1 (core)",
    "@oclif/plugin-search 0.0.17 (core)",
    "@oclif/plugin-update 3.1.15 (core)",
    "@oclif/plugin-version 1.3.4 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.37 (core)",
    "@oclif/plugin-which 2.2.20 (core)",
    "@salesforce/cli 1.79.5 (core)",
    "apex 2.2.19 (core)",
    "auth 2.7.15 (core)",
    "data 2.3.18 (core)",
    "deploy-retrieve 1.11.0 (user)",
    "info 2.6.13 (core)",
    "limits 2.3.15 (core)",
    "login 1.2.9 (core)",
    "org 2.9.0 (core)",
    "packaging 1.16.11 (user)",
    "schema 2.3.10 (core)",
    "settings 1.4.8 (core)",
    "sobject 0.1.19 (core)",
    "source 2.10.9 (core)",
    "telemetry 2.2.0 (core)",
    "templates 55.4.15 (core)",
    "trust 2.4.18 (core)",
    "user 2.3.13 (core)"
  ]
}
```

**Comment by mshanemc** [comment]
@nBray-tLP you've got a linked plugin `deploy-retrieve 1.11.0 (user)` that's old.  When you do that (installing a specific plugin version) it doesn't update when the CLI does.

Uninstall that `sf plugins uninstall deploy-retrieve` and you should be good.  While you're at it, you probably want to uninstall packaging.

Remember you can always run `sf(dx) doctor` to have a non-human catch that kind of stuff.

**Comment by ghost** [comment]
@mshanemc Thanks that's good to know! Yea, I manually updated deploy-retrieve as at the time, the one installed was older than that. when i uninstalled, it resolved back to 1.10.1. I removed packaging as well. I had only manually updated that plugin to see if it would resolve the issue.

 I continue to get my issue, though since after this update was released, it has given a different message:
```
Preparing retrieve request... done
Error (1): unknown

*** Internal Diagnostic ***

unknown: unknown
    at RetrieveMetadata.run (/usr/local/lib/node_modules/@salesforce/cli/node_modules/@salesforce/plugin-deploy-retrieve/lib/commands/project/retrieve/start.js:82:23)
    at async RetrieveMetadata._run (/usr/local/lib/node_modules/@salesforce/cli/node_modules/@oclif/core/lib/command.js:117:22)
    at async Config.runCommand (/usr/local/lib/node_modules/@salesforce/cli/node_modules/@oclif/core/lib/config/config.js:329:25)
    at async run (/usr/local/lib/node_modules/@salesforce/cli/node_modules/@oclif/core/lib/main.js:89:16)
******
```
Obviously, there's a lot more to the debug, but that's a lot to dump out here.
I get this when running:
```
sf project retrieve start --package-name ${packageName} --target-org ${userAlias} --target-metadata-dir ${dirName} --dev-debug
```

My apologies for mixing things up on the version numbers there....
As I mentioned I attempted this on the v2 beta of the sf CLI, as well as now sf version 1.79.5 and 1.80.6. I then uninstalled sf, and re-installed sfdx, and it works fine there. interesting though.

**Comment by mshanemc** [comment]
I'm having trouble replicating or even making sense of that.   It's the same plugin version for `sfdx` and `sf`? (current for `plugin-deploy-retrieve` is 1.11.0).

The `beta` is pretty much a nightly and picks up the latest plugins every time.

**Comment by ghost** [comment]
Yesterday it was 1.11, but after I uninstalled it like you mentioned, it defaulted to 1.10.1 for `sf` (both when I was on the `beta` version and for the most recent 'non-beta' version of sf) and is currently 1.9.2 for `sfdx`

**Comment by mshanemc** [comment]
@nBray-tLP sfdx should promote to a newer version today.  If you still have weird versions, please open an new gh issue so we can see all the version info the template asks for and maybe that'll help repro this.

**Comment by tggagne** [comment]
I'm getting this error a just-installed sf.  I thought I read that on May 26 the fix was was deployed.

tgagne@HWKMAC-C02DN69QMD6M dx-mga % sf version --verbose --json
{
  "cliVersion": "@salesforce/cli/2.1.7",
  "architecture": "darwin-x64",
  "nodeVersion": "node-v16.15.0",
  "osVersion": "Darwin 22.5.0",
  "shell": "zsh",
  "rootPath": "/Users/tgagne/var/node_modules/@salesforce/cli",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.2 (core)",
    "@oclif/plugin-commands 2.2.18 (core)",
    "@oclif/plugin-help 5.2.13 (core)",
    "@oclif/plugin-not-found 2.3.31 (core)",
    "@oclif/plugin-plugins 3.1.6 (core)",
    "@oclif/plugin-search 0.0.20 (core)",
    "@oclif/plugin-update 3.1.26 (core)",
    "@oclif/plugin-version 1.3.7 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.43 (core)",
    "@oclif/plugin-which 2.2.25 (core)",
    "@salesforce/cli 2.1.7 (core)",
    "apex 2.3.6 (core)",
    "auth 2.8.9 (core)",
    "data 2.5.1 (core)",
    "deploy-retrieve 1.16.0 (core)",
    "info 2.6.31 (core)",
    "limits 2.3.26 (core)",
    "login 1.2.20 (core)",
    "org 2.9.25 (core)",
    "schema 2.3.20 (core)",
    "settings 1.4.20 (core)",
    "sobject 0.1.37 (core)",
    "source 2.10.25 (core)",
    "telemetry 2.2.3 (core)",
    "templates 55.5.5 (core)",
    "trust 2.4.32 (core)",
    "user 2.3.24 (core)"
  ]
}

**Comment by LokendraSharma** [comment]
Hi @mshanemc, I'm also getting same error. Could you please shed some light to resolve this issue?
{
  "cliVersion": "@salesforce/cli/2.7.11",     
  "architecture": "win32-x64",
  "nodeVersion": "node-v18.15.0",
  "osVersion": "Windows_NT 10.0.19044",       
  "shell": "cmd.exe",
  "rootPath": "C:\\Program Files\\sf\\client",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.6 (core)",
    "@oclif/plugin-commands 2.2.24 (core)",   
    "@oclif/plugin-help 5.2.18 (core)",       
    "@oclif/plugin-not-found 2.4.0 (core)",   
    "@oclif/plugin-plugins 3.3.2 (core)",     
    "@oclif/plugin-search 0.0.22 (core)",     
    "@oclif/plugin-update 3.2.0 (core)",
    "@oclif/plugin-version 1.3.9 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.49 (core)",
    "@oclif/plugin-which 2.2.32 (core)",
    "@salesforce/cli 2.7.11 (core)",
    "apex 2.3.12 (core)",
    "auth 2.8.14 (core)",
    "data 2.5.7 (core)",
    "deploy-retrieve 1.17.6 (core)",
    "info 2.6.40 (core)",
    "limits 2.3.32 (core)",
    "login 1.2.28 (core)",
    "marketplace 0.1.1 (core)",
    "org 2.10.5 (core)",
    "schema 2.3.24 (core)",
    "settings 1.4.27 (core)",
    "sobject 0.2.6 (core)",
    "source 2.10.32 (core)",
    "telemetry 2.3.1 (core)",
    "templates 55.5.10 (core)",
    "trust 2.6.7 (core)",
    "user 2.3.31 (core)"
  ]
}

**Comment by mshanemc** [comment]
@LokendraSharma I'd prefer you to open a new issue so you can share the full command that you're running.

here's the docs for that command
https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_reference.meta/sfdx_cli_reference/cli_reference_project_commands_unified.htm#cli_reference_project_retrieve_start_unified
